src/navigator/NavigationService.ts

```diff
@@ -95,6 +96,12 @@ export function navigate<RouteName extends keyof StackParamList>(
     : [RouteName, StackParamList[RouteName]]
 ) {
   const [routeName, params] = args
+
+  if (routeName === Screens.ImportWallet) {
```

Comment:
i don't love this approach because it is a little unexpected / easy to miss. BUT, the previous implementation had this logic in the screen itself which for me was just as unexpected....with this approach at least the navigation functionality lives together and it's not possible to forget to clear the stack when navigating to this screen

Reply:
Yes though what this doesn't fully cover is the case where the app is launched and the initial screen is set to `ImportWallet`.

https://github.com/valora-inc/wallet/blob/2af220ad9b557963e35dda38f2a00fd3fb01351e/src/navigator/Navigator.tsx#L604-L607

That's why I had created the `UseBackToWelcomeScreen` and put the logic close to the screen.

That being said, without this edge case handling, I think the user is still able to proceed, just the animation is not right when tapping "cancel", after launching the app and landing directly on the import screen.

Reply:
actually this case i did test, and i think it works okay still (i haven't tested on release mode though, but i hope it doesn't make a difference). to me the animation here still looks okay, what do you think? 

https://user-images.githubusercontent.com/20150449/211284317-d2172170-ba34-45d0-9170-f4e9b4a39939.mp4

Reply:
clicking on the back button takes you out of the app though, maybe this is the issue you were referring to? i can play around with the `useBackHandler` hook for this i think

Reply:
Yes I think it's fine. The incorrect animation is more on iOS. But of course a very small detail.